### PR TITLE
Added status checking

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: controllers/cloud.redhat.com/clowdapp_controller.go [hunk 1]
```diff
@@ -142,9 +142,48 @@ func (r *ClowdAppReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
+	app.Status.ReadyDeployments = readyDeployments
+	err = proxyClient.Status().Update(ctx, &app)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{Requeue: requeue}, nil
 }
 
+func DeploymentStatusChecker(deployment *apps.Deployment) (string, bool, error) {
```
Comment (psav) [inline controllers/cloud.redhat.com/clowdapp_controller.go:171]: This came from kubectl

**Comment by kylape** [comment]
The code looks good but there are some clear opportunities to apply the DRY principle.  I wonder if we could create a new file `status.go`, create a common struct/interface that contains the deployment status, and move the `GetOwnedDeployments` function in there.  Rename `GetOwnedDeployments` to `FilterOwnedDeployment` and pass in the UID instead of having it be a member of the struct.

wdyt?  I'm not convinced it's better, but it feels like it would DRY up the code a bit.

**Comment by kylape** [comment]
Well I went ahead and tried out the refactor.  Lemme know if you think it's any better.

**Comment by psav** [comment]
Yup - think it's much better

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -9,12 +9,12 @@ rules:
 - apiGroups:
   - ""
   resources:
+  - configmaps
```
Comment (psav) [inline config/rbac/role.yaml:12]: This is a result of another change, but it's fine to leave in here too

**Comment by kylape** [comment]
@psav looks like my refactor is gone?  was that intentional?

**Pull request merged**